src/node/internal/internal_utils.ts

```diff
@@ -200,3 +208,167 @@ export function getOwnNonIndexProperties(
+export function callbackify<T1, T2, T3, TResult>(
+  fn: (arg1: T1, arg2: T2, arg3: T3) => Promise<TResult>
+): (
+  arg1: T1,
+  arg2: T2,
+  arg3: T3,
+  callback: (err: Error | null, result: TResult) => void
+) => void;
+export function callbackify<T1, T2, T3, T4>(
+  fn: (arg1: T1, arg2: T2, arg3: T3, arg4: T4) => Promise<void>
+): (
+  arg1: T1,
+  arg2: T2,
+  arg3: T3,
+  arg4: T4,
+  callback: (err: Error) => void
+) => void;
+export function callbackify<T1, T2, T3, T4, TResult>(
+  fn: (arg1: T1, arg2: T2, arg3: T3, arg4: T4) => Promise<TResult>
+): (
+  arg1: T1,
+  arg2: T2,
+  arg3: T3,
+  arg4: T4,
+  callback: (err: Error | null, result: TResult) => void
+) => void;
+export function callbackify<T1, T2, T3, T4, T5>(
+  fn: (arg1: T1, arg2: T2, arg3: T3, arg4: T4, arg5: T5) => Promise<void>
+): (
+  arg1: T1,
+  arg2: T2,
+  arg3: T3,
+  arg4: T4,
+  arg5: T5,
+  callback: (err: Error) => void
+) => void;
+export function callbackify<T1, T2, T3, T4, T5, TResult>(
+  fn: (arg1: T1, arg2: T2, arg3: T3, arg4: T4, arg5: T5) => Promise<TResult>
+): (
+  arg1: T1,
+  arg2: T2,
+  arg3: T3,
+  arg4: T4,
+  arg5: T5,
+  callback: (err: Error | null, result: TResult) => void
+) => void;
+export function callbackify<T1, T2, T3, T4, T5, T6>(
+  fn: (
+    arg1: T1,
+    arg2: T2,
+    arg3: T3,
+    arg4: T4,
+    arg5: T5,
+    arg6: T6
+  ) => Promise<void>
+): (
+  arg1: T1,
+  arg2: T2,
+  arg3: T3,
+  arg4: T4,
+  arg5: T5,
+  arg6: T6,
+  callback: (err: Error) => void
+) => void;
+export function callbackify<T1, T2, T3, T4, T5, T6, TResult>(
+  fn: (
+    arg1: T1,
+    arg2: T2,
+    arg3: T3,
+    arg4: T4,
+    arg5: T5,
+    arg6: T6
+  ) => Promise<TResult>
+): (
+  arg1: T1,
+  arg2: T2,
+  arg3: T3,
+  arg4: T4,
+  arg5: T5,
+  arg6: T6,
+  callback: (err: Error | null, result: TResult) => void
+) => void;
+export function callbackify<T extends (...args: unknown[]) => Promise<unknown>>(
+  original: T
+): T extends (...args: infer TArgs) => Promise<infer TReturn>
+  ? (...params: [...TArgs, (err: Error, ret: TReturn) => unknown]) => void
+  : never {
+  validateFunction(original, 'original');
+
+  function callbackified(
+    this: unknown,
+    ...args: [...unknown[], (err: unknown, ret: unknown) => void]
+  ): void {
+    const maybeCb = args.pop();
+    validateFunction(maybeCb, 'last argument');
+    const cb = maybeCb.bind(this);
+    Reflect.apply(original, this, args).then(
+      (ret: unknown) => {
+        queueMicrotask(() => cb(null, ret));
+      },
+      (rej: unknown) => {
+        queueMicrotask(() => {
+          callbackifyOnRejected(rej, cb);
+        });
+      }
+    );
+  }
+
+  const descriptors = Object.getOwnPropertyDescriptors(original);
+  if (typeof descriptors['length']?.value === 'number') {
+    descriptors['length'].value++;
+  }
+  if (typeof descriptors['name']?.value === 'string') {
+    descriptors['name'].value += 'Callbackified';
+  }
```

Comment:
```suggestion
  if (typeof descriptors.length?.value === 'number') {
    descriptors.length.value++;
  }
  if (typeof descriptors.name?.value === 'string') {
    descriptors.name.value += 'Callbackified';
  }
```

Reply:
Yeah, this is better, but we should definitely add an eslint rule that forbids this. I'll apply this change but there is nothing blocking others from reverting this.